zerver/lib/bugdown/__init__.py

```diff
@@ -144,7 +144,12 @@ def add_a(root, url, link, title=None, desc=None,
     if data_id is not None:
         a.set("data-id", data_id)
     img = markdown.util.etree.SubElement(a, "img")
-    img.set("src", url)
+    thumbnail_path = '/thumbnail'
+    if not url.startswith('/'):
+        thumbnail_path += '/'
```

Comment:
I believe this should be the right way. Using `urljoin` doesn't give an expected behaviour here always. What I am doing here is that any incoming link that will have a preview image will have a img src set to call the `/thumbnail` endpoint. 
Now using `urljoin` creates problems here since it joins relatively and doesn't full fill expectations here.